### PR TITLE
Inject user to middleware tests

### PR DESCRIPTION
**Pull request by stsewd**
We don't use the full middleware cycle,
so we don't have a `request.user`.

This fixes the tests failing in the corporate site

### CODE REVIEW

File: readthedocs/rtd_tests/tests/test_middleware.py [hunk 1]
```diff
@@ -137,15 +139,19 @@ def test_use_subdomain(self):
...
 
+    @override_settings(PRODUCTION_DOMAIN='readthedocs.org')
     def test_long_bad_subdomain(self):
         domain = 'www.pip.readthedocs.org'
         request = self.factory.get(self.url, HTTP_HOST=domain)
+        request.user = self.owner
         res = self.middleware.process_request(request)
         self.assertEqual(res.status_code, 400)
 
+    @override_settings(PRODUCTION_DOMAIN='readthedocs.org')
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_middleware.py]: This was failing bc we override `production_domain`  to `readthedocs.com`

File: readthedocs/rtd_tests/tests/test_middleware.py [hunk 2]
```diff
@@ -32,6 +32,7 @@ def setUp(self):
     def test_failey_cname(self):
         self.assertFalse(Domain.objects.filter(domain='my.host.com').exists())
         request = self.factory.get(self.url, HTTP_HOST='my.host.com')
+        request.user = self.owner
```
Comment (agjohnson) [inline readthedocs/rtd_tests/tests/test_middleware.py]: Would setting `request.user = AnonymousUser()` be closer to the original test case?

Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_middleware.py]: Yeah, I think so, I retreated from using it bc of another failure but in the end was unrelated, I'll test again.

Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_middleware.py]: AnonymousUser works!

**Pull request merged**